src/controller/CHIPDevice.cpp

```diff
@@ -244,9 +244,9 @@ CHIP_ERROR Device::LoadSecureSessionParameters(ResetTransport resetNeeded)
 
 bool Device::GetIpAddress(Inet::IPAddress & addr) const
 {
-    if (mState == ConnectionState::SecureConnected)
+    if (mState != ConnectionState::NotConnected)
```

Comment:
@pan-apple Could you double-check this, please.

Reply:
not: I would peronsally find this easier to read:

```
if (mState == ConnectionState::NotConnected) {
  return false;
}

addr = mDeviceAddr;
return true;
```

this shows that 'return true' and 'set device addr' are really closely tied together.